docs/hints-for-use-with-docker.md

```diff
@@ -0,0 +1,52 @@
+# Hints for using ORT with Docker
+
+- [Hints for using ORT with Docker](#hints-for-using-ort-with-docker)
+  - [General Hints](#general-hints)
+    - [Mount current working directory or directory relative to working directory](#mount-current-working-directory-or-directory-relative-to-working-directory)
+  - [Common Issues](#common-issues)
+    - [Docker build fails with a "SSL Handshake" error](#docker-build-fails-with-a-ssl-handshake-error)
+    - [Authenticating with a private git repository fails](#authenticating-with-a-private-git-repository-fails)
+
+## General Hints
+
+### Mount current working directory or directory relative to working directory
+
+Use `$(pwd)` on unix-based systems, `${PWD}` for PowerShell on Windows:
+
+```shell
+docker run \
+  -v $(pwd):/project  \ # mount current working directory into /project to use as input
+  -v $(pwd).ort:/ort \ # mount folder "<workingdirectory>.ort" to /ort to use as output and config file storage
+  ort --info analyze \
+  -c /ort/config.hocon \ # use file from "<workingdirectory>.ort" as config
+  analyze (...) # further options
+```
+
+## Common Issues
+
+### Docker build fails with a "SSL Handshake" error
+
+1. Insert `ENV JAVA_OPTS="-Djdk.tls.client.protocols=TLSv1.2"` in the Dockerfile, below the `FROM` line to force a specific TLS version.
+2. Run the build again, it should succeed now.
+
+### Authenticating with a private git repository fails
+
+To authenticate with a private git repository, ORT uses the (semi)standardized .netrc file. With the following steps, `.netrc` can be used with Docker.
+
+1. Create a .netrc file:
+   ```
+   machine <hostname> login <username> password <password>
+   ```
+   For example:
+   ```
+   machine github.com login mygituser password mypersonaltoken
+   ```
+   Ensure that there are no additional spaces, newlines, tabs, or other whitespace characters after the hostname, username or password. These characters would be appended to the information and cause authentication errors for some providers.
+
+   **Note:** If you receive any authentication errors, double-check this file. The format often shared online with the properties separated by newlines does **not** work with Docker images built from ORTs alpine-linux based Dockerfile.
```

Comment:
s/alpine-linux/Alpine Linux/

But does it actually matter here that the Docker image is Alpine Linux based? If not, we should drop that bit.

Also, I'm surprised that you say the newline separated `.netrc` file would not work, as we even have a text for it, see

https://github.com/oss-review-toolkit/ort/blob/c480f13d5d9fe14af8de6bf98c4954f77a6785b6/utils/src/test/kotlin/OrtAuthenticatorTest.kt#L39-L49

Would you mind elaborating?

Reply:
**ad Alpine Linux**: As discussed in the other thread regarding this, native tools also struggle with the format, but apparently this is depending on the distribution-specific implementation (at least for cURL), hence the hint. But I can drop it, let someone tries 10 different docker image bases, before simply adjusting their `.netrc`.

**ad `.netrc`**: I assume the test case only handles the manual parsing, not the native tools. I tested this running the docker image both on Windows 10 and Ubuntu 20.04 (should not make a difference, but for reference) and only the `.netrc` without newlines worked. The same also reproduced running Git natively in Ubuntu 20.04 without involving ORT at all, simply cloning. 

This may also depend on the repository hosting provider: GitHub definitely does not strip whitespace characters included in the Authentication HTTP header on their end, leading to a HTTP 401 error due to wrong username/password, but other providers may handle this differently and avoid errors.